Bosworth Toller's Anglo-Saxon Dictionary Online.js

```diff
@@ -0,0 +1,636 @@
+	// form in the print book) can be found either in the metadata or in the
+	// "citation" block under "entry information" in the doc, no matter the
+	// citation style in use on the page.
+	let titleRaw = text(doc, "#btd--entry-lemma").trim();
+	let titleNoDiac = removeDiac(titleRaw);
+
+	const metaTitle = attr(doc, 'meta[property="og:title"]', "content").trim();
+	const metaNoDiac = removeDiac(metaTitle);
+
+	if (metaTitle && titleNoDiac === metaNoDiac) {
+		return metaTitle;
+	}
+
+	// Fallback, using the citation line, just in case.
+	if (!titleRaw) {
+		titleRaw = metaTitle;
+		titleNoDiac = metaNoDiac;
+	}
+	const citeRaw = ZU.trimInternal(text(doc, ".btd--entry-information p"));
+	const citeNoDiac = removeDiac(citeRaw);
+
+	if (!titleNoDiac || !citeNoDiac) {
+		return null;
+	}
+
+	// Retrieve the index of the lemma, modulo diacritics, in the
+	// de-diacritized citation line.
+	const idx = citeNoDiac.search(titleNoDiac);
+	if (idx === -1) {
+		Z.debug(`Error: entry-form "${titleRaw}" not found in citation line: ${citeRaw}`);
+		return null;
+	}
+	// Use the index to retrieve the corresponding slice in the original
+	// citation line.
+	// NOTE: We can get away with the following, because (hopefully) we're not
+	// running into any diacritized letters, or even astrals, before the end of
+	// the lemma in the citation line...
+	return citeRaw.slice(idx, idx + titleRaw.length);
+}
+
+// Return an array of a single Note object representing the word definitions,
+// word-category info, and the grammar info, or false if none of such info is
+// found.
+function getNotes(doc, title) {
+	// We do so by creating a detached DOM subtree at "root" using the text
+	// info gleaned from the page. We will only use plain-text info scraped
+	// from the doc to construct our own subtree; no node from the doc, apart
+	// from its textContent, is directly used. This is done so as to return a
+	// minimal HTML-fragment usable as the note, with no attributes etc.
+	// Once done, the subtree's innerHTML (which contains only text nodes and
+	// elements we created on our own, with sanitized input) is used as the
+	// source of the Note.
+	const root = doc.createElement("div");
+	let found = false;
+
+	// Make an element with enclosed text content (must not be a void element).
+	function elem(tag, content) {
+		const newElem = doc.createElement(tag);
+		if (content) {
+			newElem.textContent = content;
+		}
+		return newElem;
+	}
+
+	// Add a section under the given heading level and text, and if "below" is
+	// given, append it (i.e. it becomes the content below the heading).
+	function addSection(headingLevel, headingText, below) {
+		root.append(elem(headingLevel, headingText));
+		if (below) {
+			root.append(below);
+		}
+		return root;
+	}
+
+	// Make an <ul> element, with given array of strings as the text content
+	// in its <li> child elements.
+	function itemize(arr) {
+		const list = doc.createElement("ul");
```

Comment:
no strong feeling, but the definitions in the dictionary are numbered, aren't they? So why not use `ol`?